pipenv/cli.py

```diff
@@ -1762,11 +1762,7 @@ def install(
         if package_names[0]:
             if not package_names[0].startswith('-e '):
                 if not is_file(package_names[0]):
-                    if not (
-                        ('==' in package_names[0]) or
-                        ('>=' in package_names[0]) or
-                        ('<=' in package_names[0])
-                    ):
+                    if not any(op in package_names[0] for op in '!=<>~'):
```

Comment:
I don't think this is very good.

foo=1.2.3 would satisfy this, but is invalid.  Using `=` instead of `==` is a pretty common mistake I think.

What's wrong with

    if not any(op in package_names[0] for op in {'==', '>=', '<=', '!=', '~=', '<', '>', '==='}):

It's more verbose, but much more accurate.

Reply:
@darkfeline, I'm not entirely sure I understand your statement. The case you're proposing was unnecessarily being checked before, resulting in a longer processing time. This check is only for suggesting a better package name. I have no qualms and actually think it's an improvement that `foo=1.2.3` won't make it through this conditional any more. We have other code that handles inappropriate version specifiers and the checks can remain handled at that level.

Reply:
> This check is only for suggesting a better package name

Ah, I see.  That makes sense then.

Reply:
`{'==', '>=', '<=', '!=', '~=', '<', '>', '==='}`
This is a very simple change, but quite a step forward - it makes user think what did she done wrong, if she did. I would have changed it if the pr hadn't been already merged.

But I think this part should be redesigned anyway. If user fails to pass correct specifier, pip explodes with a stacktrace. I think that it would be good to analyze input more precisely and show errors in human-acceptable way. Maybe we want to split an expression into tokens, and show error+suggestion for every token, I don't know